modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java

```diff
@@ -89,86 +157,468 @@ public IgniteStatisticsRepository statisticsRepository() {
     }
 
     /** {@inheritDoc} */
-    @Override public void clearObjectStatistics(String schemaName, String objName, String... colNames) {
-        StatsKey key = new StatsKey(schemaName, objName);
+    @Override public void clearObjectStatistics(
+            String schemaName,
+            String objName,
+            String... colNames
+    ) throws IgniteCheckedException {
+        final StatsCollectionFutureAdapter doneFut = new StatsCollectionFutureAdapter();
+
+        UUID reqId = UUID.randomUUID();
+        StatsKeyMessage keyMsg = StatisticsUtils.toMessage(schemaName, objName, colNames);
+
+        Map<UUID, List<Integer>> requestNodes = null;
+        /*try {
+            requestNodes = nodePartitions(extractGroups(Collections.singletonList(keyMsg)), null);
+        } catch (IgniteCheckedException e) {
+            // TODO: handle & remove task
+        }
+
+        StatsClearRequest req = new StatsClearRequest(reqId, false, Collections.singletonList(keyMsg));
+
+        sendLocalRequests(reqId, req, requestNodes);*/
+
+        clearObjectStatisticsLocal(keyMsg);
+    }
+
+    /**
+     * Actually clear local object statistics by the given key.
+     *
+     * @param keyMsg Key to clear statistics by.
+     */
+    private void clearObjectStatisticsLocal(StatsKeyMessage keyMsg) {
+        StatsKey key = new StatsKey(keyMsg.schema(), keyMsg.obj());
+        String[] colNames = keyMsg.colNames().toArray(new String[0]);
+
         statsRepos.clearLocalPartitionsStatistics(key, colNames);
         statsRepos.clearLocalStatistics(key, colNames);
         statsRepos.clearGlobalStatistics(key, colNames);
     }
 
+    // TODO
+    /**
+     * Collect local object statistics by primary partitions of specified object.
+     *
+     * @param keyMsg Statistic key message to collect statistics by.
```

Comment:
Statistic is collected by a 'key' not by a 'message', right?

Reply:
No, we just store and aquire statistics by whole object (by StatsKey), but we can collect and clear it with column granularity (by StatsKeyMessage).